Cart Flow/Custom Views/LocationSelectionViews/AisleSelectionView.swift

```diff
@@ -0,0 +1,173 @@
+//
+//  AisleSelectionVC.swift
+//  Cart Flow
+//
+//  Created by Scott Quintana on 8/25/20.
+//  Copyright © 2020 Scott Quintana. All rights reserved.
+//
+
+import UIKit
+import CoreData
+
+protocol AisleSelectionViewDelegate: class {
+    func presentAisleAlert(alert: UIAlertController)
+}
+
+class AisleSelectionView: UIView {
+    
+    
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    
+    let aislesLabel = CFSecondaryTitleLabel()
+    let aisleDescriptionLabel = CFSecondaryTitleLabel()
+    let plusButton = CFPlusButton()
+    let minusButton = CFMinusButton()
+    
+    let aislePicker = UIPickerView()
+    
+    var aisles: [Aisle] = []
+    var selectedStore: GroceryStore? = nil
+    var selectedAisle: Aisle? = nil
+    var rotationAngle: CGFloat!
+    
+    weak var delegate: AisleSelectionViewDelegate!
+    
+    
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        
+        configureAislePicker()
+        layoutUI()
+    }
+    
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    private func configureAislePicker() {
+        self.aislePicker.delegate = self
+        self.aislePicker.dataSource = self
+    }
+    
+    
+    private func layoutUI() {
+        addSubview(aislesLabel)
+        addSubview(aisleDescriptionLabel)
+        addSubview(aislePicker)
+        
+        let padding: CGFloat = 15
+        rotationAngle = 90 * (.pi/180)
```

Comment:
While this is a clever way to do this, I'd caution against modifying standard iOS controls this heavily. If you need, you can pull in an external library to provide this view (check out https://www.cocoacontrols.com for some custom views you can use, and [CocoaPods](https://cocoapods.org) or [Carthage](https://github.com/Carthage/Carthage) for dependency management) or find something that's more fitting from the stock controls in iOS. In general, make sure you're following [Apple's Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/).

Reply:
These pickers have been such a pain for me, and I still don't love how they function from a UX standpoint. Will check out some libraries. Great idea.

Reply:
Yeah, I think the way you did it makes sense, but I just think the UI might get weird on older or future versions of iOS. Another way to tackle this is to push to a new screen with a table of aisles for selecting the aisle instead of doing it inline on the same screen.

Reply:
It already got weird with the iOS14 upgrade, hah. I think I have an idea to rework it. Will give me a chance to redo the MVC violations as well!